bzt/modules/selenium.py

```diff
@@ -1350,8 +1366,55 @@ def gen_assertion(self, assertion_config):
                 assertion_elements.append(self.gen_statement(method))
         return assertion_elements
 
-    def __gen_assert_page(self):
-        return self.gen_statement("body = self.driver.page_source")
+    def gen_action(self, action_config):
+        aby, atype, param, selector = self._parse_action(action_config)
+
+        bys = {
+            'byxpath': "XPATH",
+            'bycss': "CSS_SELECTOR",
+            'byname': "NAME",
+            'byid': "ID",
+        }
+        if atype in ('click', 'keys'):
+            tpl = "self.driver.find_element(By.%s, %r).%s"
+            if atype == 'click':
+                action = "click()"
+            else:
+                action = "send_keys(%r)" % param
+
+            return self.gen_statement(tpl % (bys[aby], selector, action))
+        elif atype == 'wait':
+            tpl = "WebDriverWait(self.driver, %s).until(econd.%s_of_element_located((By.%s, %r)), %r)"
+            mode = "visibility" if param == 'visible' else 'presence'
+            exc = TaurusInternalException("Timeout value should be present")
+            timeout = dehumanize_time(self.scenario.get("timeout", exc))
+            errmsg = "Element %r failed to appear within %ss" % (selector, timeout)
+            return self.gen_statement(tpl % (timeout, mode, bys[aby], selector, errmsg))
+
+        raise TaurusInternalException("Could not build code for action: %s" % action_config)
+
+    def _parse_action(self, action_config):
+        if isinstance(action_config, string_types):
+            name = action_config
+            param = None
+        elif isinstance(action_config, dict):
+            name = list(action_config.keys())[0]
```

Comment:
Oh, i see, it's for cases like `waitByName`, where it's a one-element dictionary.

Using something like `name, param = next(iteritems(action_config))` might look a little easier on the eyes.